server.go

```diff
@@ -90,15 +90,13 @@ func NewServer(be Backend) *Server {
 						return errors.New("Identities not supported")
 					}
 
-					state := conn.State()
-					session, err := be.Login(&state, username, password)
-					if err != nil {
-						return err
+					sess := conn.Session()
+					if sess == nil {
+						panic("No session when AUTH is called")
 					}
 
-					conn.SetSession(session)
-					return nil
-				})
+					return sess.AuthPlain(username, password)
+					})
```

Comment:
Indentation is off here

Reply:
[Fixed](https://github.com/emersion/go-smtp/compare/ec3cfb6648fb2d78cdc8907c75a60970e7f69333..03f13880e866f1351daca7d62d22936ea3d1a6c7)